src/shared/types.js

```diff
@@ -4,18 +4,57 @@ import type {
     Logger as WinstonLogger,
     Info as WinstonInfo,
 } from "winston";
+
 import type {$Request} from "express";
 
 export type Info = WinstonInfo<NpmLogLevels>;
+
 export type LogLevel = $Keys<NpmLogLevels>;
+
 export type Logger = WinstonLogger<NpmLogLevels>;
+
+/**
+ * Information to attach to a trace session.
+ */
+export type TraceSessionInfo = {
+    level?: LogLevel,
+    [datum: string]: mixed,
+    ...
+};
+
+/**
+ * A trace session that has been started.
+ */
+export interface ITraceSession {
+    /**
+     * The name of the session as provided when it was started.
+     */
+    get name(): string;
```

Comment:
The RRS version of a session did not include this, but I realised it may be useful to have, especially if we make this API available to more things and we want to include additional state info like whether the trace is ended yet, and if it is ended, how long it took.  For now, this should help with debugging at the very least.